build.sbt

```diff
@@ -106,7 +109,8 @@ lazy val commonJsSettings = Seq(
 
 lazy val commonJvmSettings = Seq(
   parallelExecution in Test := false,
-  coverageExcludedPackages := "shapeless.examples.*"
+  coverageExcludedPackages := "shapeless.examples.*",
+  skip.in(publish) := customScalaJSVersion.forall(_.startsWith("0.6"))
```

Comment:
This setting should prevent publishing the jvm side twice. I think native is called explicitly on the build and not as part of root

Reply:
I missed that. It still builds and tests non-js twice though, doesn't it?

Reply:
Yes, that is true, I think CI times probably extended due to that change. I wonder if we could at least skip tests in the same way

Reply:
Perhaps it would be better to separate js and non-js builds completely?

Reply:
Monocle takes that approach with two 'root' projects for JVM and JS

Reply:
Can we not do `coreJVM/publish` vs. `coreJS/publish`?